src/pages/CreateBoard/index.tsx

```diff
@@ -5,35 +5,47 @@ import theme from 'styles/theme';
 import Input from 'components/Input';
 import UploadIcon from 'assets/UploadIcon';
 import { useState } from 'react';
-import instance from 'apis/httpClient';
+import useImageUpload from 'hooks/useImageUpload';
+import { useMutation, useQueryClient } from 'react-query';
+import { createBoard } from 'apis/api';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
 
 const CreateBoard = () => {
-  const [imageUrl, setImageUrl] = useState<string>('');
+  const [name, setName] = useState<string>('');
+  const [content, setContent] = useState<string>('');
+  const { imageUrl, handleImageChange } = useImageUpload();
+  const router = useNavigate();
 
-  const handleImageChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
-    if (e.target.files && e.target.files.length > 0) {
-      const selectedFile = e.target.files[0];
-      setImageUrl(URL.createObjectURL(selectedFile));
+  const onChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const {
+      target: { name, value },
+    } = e;
 
-      const formData = new FormData();
-      formData.append('image', selectedFile);
+    if (name === 'title') {
+      setName(value);
+    }
+
+    if (name === 'content') {
+      setContent(value);
+    }
+  };
 
-      const blob = new Blob([JSON.stringify(selectedFile)], {
-        type: 'application/json',
-      });
+  const queryClient = useQueryClient();
 
-      formData.append('data', blob);
+  const { mutate } = useMutation({
+    mutationFn: () => createBoard({ title: name, content, imgUrl: imageUrl }),
+    onSuccess: () => {
+      queryClient.invalidateQueries(['readBoard']);
+    },
+  });
 
-      try {
-        const { data } = await instance.post('/project/image', formData, {
-          headers: {
-            'Content-Type': 'multipart/form-data',
-          },
-        });
-      } catch (error) {
-        console.error('Error uploading image:', error);
-      }
-    }
+  const onSubmit = () => {
+    mutate();
```

Comment:
```suggestion
    postBoardMutate();
```
와 같이 추상화를 해주면 개발자 위에 코드를 스크롤해서 읽어 보지 않아도 되니깐 경험이 향상될것같아요!

Reply:
감사합니다